sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -557,8 +557,11 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
             "TOK_TABLEPROPERTIES"),
           children)
       val (db, tableName) = extractDbNameTableName(tableNameParts)
-
-      CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, Some(node))
+      if (ignores.exists(_ != None)) {
+        CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, Some(node))
+      } else {
+        CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, None)
```

Comment:
This part is tricky. The name of `hasStorageSpec` is not very accurate. Actually, when text file and sequence file is used (set by Hive), `crtTbl.getSerName` will be `null`. My intension is when `convertCTAS` is true and text file or sequence file is used, we will use parquet (basically convertCTAS is used to avoid saving data to an inefficient format). If the setting of `hive.default.fileformat` is any other format, we will use that format. You can refer to my comment above and 
the comment of convertCTAS (https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala#L72) for more information.